packages/aws-event-stream/src/aws_event_stream/_private/serializers.py

```diff
@@ -36,16 +36,23 @@
 _DEFAULT_BLOB_CONTENT_TYPE = "application/octet-stream"
 
 
-type Signer = Callable[[EventMessage], EventMessage]
-"""A function that takes an event message and signs it, and returns it signed."""
+class EventSigner(Protocol):
+    """A signer to manage credentials and EventMessages for an Event Stream lifecyle."""
+
+    def sign_event(
```

Comment:
I'd like to move this to the regular signer. We can create a central event dataclass or protocol to give it. That should solve the double initialization problem. That can be later though.

Reply:
We need to track signing state (previous signature) between events so theres some state management.  But we could also add previous signature as an input to sign_event and store it in the publisher.

Reply:
Yeah my original plan had been for this context to be in the Publisher rather than the signer. What we have implemented currently makes that a bit harder. I think raising it a level to let the publisher do the orchestration makes more sense long term though.

Reply:
We can do that without muddying up the interface too much. Just make an `event_signer` method on the auth scheme class or whatever that's given the entire transport request and returns the signer to use. A stateless signer could just return itself.